### PR TITLE
MOTECH-1710: ScheduleTracking javadoc service package

### PR DESCRIPTION
**Pull request by tstalka**

### CODE REVIEW

File: schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/EnrollmentRequest.java [hunk 1]
```diff
@@ -11,19 +11,27 @@
 import static org.motechproject.commons.date.util.DateUtil.today;
 
 /**
- * \ingroup sts
- *
- * This is the request document to create an enrollment
+ * <code>EnrollmentRequest</code> represents the request document to create an enrollment.
```
Comment (pgesek) [inline schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/EnrollmentRequest.java]: Document is an leftover term from couchDB days.

File: schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/EnrollmentService.java [hunk 2]
```diff
@@ -6,12 +6,58 @@
 
 import java.util.Map;
 
+/**
+ * Enrollment Service interface provides methods for enroll and unenroll enrollments, fulfill milestones and for retrieving
```
Comment (pgesek) [inline schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/EnrollmentService.java]: for enrolling and unenrolling, fullfiling milestones..

File: schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/EnrollmentService.java [hunk 3]
```diff
@@ -6,12 +6,58 @@
...
 public interface EnrollmentService {
-    Long enroll(String externalId, String scheduleName, String startingMilestoneName, DateTime referenceDateTime, DateTime enrollmentDateTime, Time preferredAlertTime, Map<String, String> metadata);
 
+    /**
+     * Creates the enrolment from the given parameters and schedule alerts jobs. This method also sends
+     * {@link org.motechproject.scheduletracking.events.EnrolledUserEvent} if there is no active enrolment for the
+     * external id and the schedule name. If the user has already enrolled for the same schedule which is currently
+     * active, then enrollment is updated and alerts are rescheduled.
+     *
+     * @param externalId the client external id
```
Comment (pgesek) [inline schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/EnrollmentService.java:21]: what client? it's the external id used to identify the subject of this enrollment (patient, client, whatever)

Applies to all external id params

File: schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/EnrollmentService.java [hunk 4]
```diff
@@ -6,12 +6,58 @@
...
+     * @param scheduleName the name of the schedule
+     * @param startingMilestoneName the starting milestone name
+     * @param referenceDateTime the reference date and time
```
Comment (pgesek) [inline schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/EnrollmentService.java]: The javadoc should shed more light on the significance of these 3 datetime parameters

Applies to all methods that have  these params

File: schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/EnrollmentService.java [hunk 5]
```diff
@@ -6,12 +6,58 @@
...
+     * @param metadata the metadata
+     * @return the id of the created enrollment
+     */
+    Long enroll(String externalId, String scheduleName, String startingMilestoneName, DateTime referenceDateTime,
+                DateTime enrollmentDateTime, Time preferredAlertTime, Map<String, String> metadata);
+
+    /**
+     * Fulfills the current milestone of the enrollment(with fulfillment date and time as the the given date and time).
+     *
+     * @param enrollment the enrollment with milestone to fulfills
```
Comment (pgesek) [inline schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/EnrollmentService.java]: to fulfill

File: schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/EnrollmentService.java [hunk 6]
```diff
@@ -6,12 +6,58 @@
...
+     * Unenrolls the given enrollment and unschedules all alerts job of the enrollment. This method also sends
+     * {@link org.motechproject.scheduletracking.events.UnenrolledUserEvent}.
+     *
+     * @param enrollment the enrollment to unenroll
+     */
     void unenroll(Enrollment enrollment);
 
-    MilestoneAlerts getAlertTimings(String externalId, String scheduleName, String milestoneName, DateTime referenceDateTime, DateTime enrollmentDateTime, Time preferredAlertTime);
+    /**
+     * Returns the alerts timings for the given parameters. Enrollment is build from the parameters and then alerts timing
```
Comment (pgesek) [inline schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/EnrollmentService.java]: alert timings

File: schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/ScheduleTrackingService.java [hunk 7]
```diff
@@ -8,97 +8,119 @@
...
+ * in a schedule and uneroll an external id from a schedule. It also provides querying functionality on Enrollments
+ * using various search criteria.
  */
 public interface ScheduleTrackingService {
+
     /**
-     * Enrolls a user with external id into a schedule using the details in the EnrollmentRequest and schedules alerts. If the user has already enrolled for the same schedule which is currently active, then we update the existing details and reschedule alerts
-     * @param enrollmentRequest
-     * @return enrollmentId string
+     * Enrolls a user with external id into a schedule using the details in the EnrollmentRequest and schedules alerts.
```
Comment (pgesek) [inline schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/ScheduleTrackingService.java]: Regarding my earlier comment about clients and external ids, I guess we can settle for users as the term used throughout the module.

File: schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/impl/EnrollmentAlertService.java [hunk 8]
```diff
@@ -27,10 +27,14 @@
 
 import static org.motechproject.commons.date.util.DateUtil.now;
 
+/**
+ * Component used for manage alerts.
```
Comment (pgesek) [inline schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/impl/EnrollmentAlertService.java]: managing

File: schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/impl/EnrollmentsQueryService.java [hunk 9]
```diff
@@ -10,6 +10,9 @@
 import java.util.ArrayList;
 import java.util.List;
 
+/**
+ * Component used for searching enrollments.
```
Comment (pgesek) [inline schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/impl/EnrollmentsQueryService.java]: Service

File: schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/impl/EnrollmentDefaultmentService.java [hunk 10]
```diff
@@ -16,16 +16,27 @@
 import static org.motechproject.commons.date.util.DateUtil.now;
 import static org.motechproject.scheduletracking.events.constants.EventSubjects.MILESTONE_DEFAULTED;
 
+/**
+ * Component to manage defaulted enrollments.
```
Comment (pgesek) [inline schedule-tracking/src/main/java/org/motechproject/scheduletracking/service/impl/EnrollmentDefaultmentService.java]: Service, not component

**Comment by pgesek** [comment]
See inline comments

**Comment by tstalka** [comment]
Test this please

**Comment by tstalka** [comment]
Test this please

**Comment by pgesek** [comment]
test this please

**Pull request merged**